diff --git a/sources/Muxing/VideoOutputStream.cpp b/sources/Muxing/VideoOutputStream.cpp
index 5a555a4..ca94678 100644
--- a/sources/Muxing/VideoOutputStream.cpp
+++ b/sources/Muxing/VideoOutputStream.cpp
@@ -5,14 +5,14 @@ using namespace std;
 
 namespace ffmpegcpp
 {
-	VideoOutputStream::VideoOutputStream(Muxer* muxer, Codec* codec)
-		: OutputStream(muxer, codec)
+	VideoOutputStream::VideoOutputStream(Muxer* p_muxer, Codec* p_codec)
+		: OutputStream(p_muxer, p_codec)
 	{
 	}
 
-	void VideoOutputStream::OpenStream(AVStream* stream, int containerFlags)
+	void VideoOutputStream::OpenStream(AVStream* p_stream, int containerFlags)
 	{
-		this->stream = stream;
+		this->stream = p_stream;
 
 		// special case for certain containers
 		if (containerFlags & AVFMT_GLOBALHEADER)
